tests/02_generic_input/field_python_test.py

```diff
@@ -0,0 +1,25 @@
+import flowpy
+
+class PyFieldAnisotropy(flowpy.PythonFieldBase):
+    pass
+
+    def used_fields(self):
+        field_list = ["X"]
+        return field_list
+    
+    def __call__(self):
+        diag = (0.5+2*z if (z < 0.25) else 1)
+        return ( diag, 0, 0, 0, diag, 0, 0, 0, diag )
+
+
+class PyFieldWaterSourceDensity(flowpy.PythonFieldBase):
+    pass
+
+    def used_fields(self):
+        field_list = ["X"]
+        return field_list
+    
+    def __call__(self):
+        x = self.X[0]
+        y = self.X[1]
+        return ( 2*(1-x**2)+2*(1-y**2), )
```

Comment:
Use vector syntax:
```
XY = self.X[0:2]
AB = 2*(1-XY**2)
return AB[0] + AB[1]   # or np.sum(AB, axis=0)
```

Reply:
Tests for more complex calculations that may be relevant:

` return np.linalg.norm(A @ X, axis = 0)`